### DISCUSSION TITLE
Expected performance on a Raspberry Pi 3 Model B+

### DISCUSSION THREAD

**Discussion by ofalvai** [firefly-iii/firefly-iii#9085] [Firefly III]
I'm running Firefly on a Raspberry Pi and most page loads take multiple seconds. Performing heavier operations (bulk edits) take multiple minutes. I tried migrating from SQLite to Postgres, but the performance didn't improve.

So I'm wondering what is the baseline I should expect when running on a Raspberry Pi. Is there anything wrong with my setup that I should look into, or is this the state of things?

**Comment by JC5** [comment]
You can track the Firefly III logs to see what's taking so long, it shouldn't be several minutes, not even on a 3B+

https://docs.firefly-iii.org/how-to/general/debug/

**Comment by ofalvai** [comment]
Thank you, this revealed the root cause and that I am dumb. Turns out, Firefly spent 10 seconds processing webhooks, and when I checked my webhook settings, I found an old integration that I set up. The connected service is not even running anymore, so Firefly was waiting 10 seconds for the webhook to time out.
